app/controllers/team_members_controller.rb

```diff
@@ -1,42 +1,50 @@
 class TeamMembersController < ApplicationController
 
-  def edit
-    @team_member = TeamMember.find(params[:id])
+  before_action :set_team_member
 
+  def edit
     if @team_member.role == 'custodian'
       @team_members = @team_member.team.team_members.joins(:user).where.not(users: {full_name: ''} )
     end
   end
 
   def update
-
-    team_member = TeamMember.find(params[:id])
-
-    if team_member.role == 'custodian'
+    if @team_member.role == 'custodian'
       if params[:new_custodian_team_member_id] == params[:id]
         flash[:notice] = 'Your custodian remains unchanged'
       else
         new_custodian = TeamMember.find(params[:new_custodian_team_member_id])
 
-        team_member.role = 'advanced'
+        @team_member.role = 'advanced'
         new_custodian.role = 'custodian'
 
-        team_member.save
+        @team_member.save
         new_custodian.save
 
         flash[:notice] = 'The custodian has been changed'
       end
-      redirect_to team_path(team_member.team_id)
+      redirect_to team_path(@team_member.team_id)
     else
-      if team_member.update(role: params[:team_member][:role])
+      if @team_member.update(role: params[:team_member][:role])
         flash[:notice] = 'Your update has been successful'
-        redirect_to team_path(team_member.team_id)
+        redirect_to team_path(@team_member.team_id)
       else
         flash[:notice] = 'Your update has failed'
         render :edit
       end
     end
+  end
+
+  def destroy
+    @team_member.destroy
+    flash[:notice] = "Successfully removed #{@team_member.user.full_name} from team"
+    redirect_to team_path(@team_member.team_id)
+  end
 
+  private
+
+  def set_team_member
```

Comment:
Personally I think this is a REALLY bad practise, I was thinking about taking it out of the code in other places. I think its a bad pattern for a number of reasons:
1. someone reading will assume the `@team_member` came from the page submit
2. Methods should be self contained 
3. You are destroying the security of your model to remove 3 lines of code (@ = instance)
4. uses more memory because the object is always around, not only when you need it
5. It makes your tests bad as you are not testing 1 thing (a unit) anymore
6. Depending on another method makes your code more fragile
7. its harder to debug
8. I think its not what before is for, before is the shorthand for `before_filter`, its meant to be a filter like checking some is logged in before allowing a method to run or checking a change has been approved before submitting it.
9. in this case you haven't even saved lines, you removed 1 line from edit and 1 from update and added a method that is 3 lines and a before statement which is another. 

I could be wrong, its just what I see when I look at it. What do you think?